packages/peregrine/src/Router/resolveUnknownRoute.js

```diff
@@ -3,25 +3,39 @@
  * with the assigned Root Component (and its owning chunk) from the backend
  * @param {{ route: string, apiBase: string, __tmp_webpack_public_path__: string}} opts
  */
+
+// Some M2.3.0 GraphQL node IDs are numbers and some are strings, so explicitly
+// cast numbers if they appear to be numbers
```

Comment:
Shouldn't this sort of interp happen in the API? I assumed the HTTP request stringifies anyways so I guess my question is why we need to coerce to a number at all in the client side.

Reply:
We switched from this gross hand-built JSON (since UPWARD doesn't serialize JSON) into using HTML data attributes to preload the URL resolver stuff. All HTML attributes are strings, so we need to coerce to the right variable types for the query--and _presently_, all our category and CMS pages have `Int` IDs. But, @paliarush has told me that the ID field might become a string, or even the native GraphQL ID type, in the future. So this is a way to allow for that future possibility, while casting recognizable numbers today.

Reply:
> we need to coerce to the right variable types for the query

And this is required because we form the query (at which point type matters) on the client side?

I guess I'm just confused because in HTTP requests `param=123` is a string and the casting happens on the server.

Reply:
Exactly. Apollo creates the query after picking up the urlResolver details.